test/repository/TpsRepoSpec.scala

```diff
@@ -46,6 +47,24 @@ class TpsRepoSpec extends ItSpec {
     result.n shouldBe 1
   }
 
+  "findPaymentItem should optionally find the matching payment item" in {
```

Comment:
I wonder why this has not been tested in IT test in `TpsControllerSpec` instead. There the test could cover application behaviour from client perspective (404 and 500 status codes and body messages for erroneous cases).

Reply:
Will re-factor